src/components/ModelingSidebar/ModelingPanes/FeatureTreePane.tsx

```diff
@@ -0,0 +1,393 @@
+import { Diagnostic } from '@codemirror/lint'
+import { useMachine, useSelector } from '@xstate/react'
+import { ContextMenu, ContextMenuItem } from 'components/ContextMenu'
+import { CustomIcon, CustomIconName } from 'components/CustomIcon'
+import Loading from 'components/Loading'
+import { useModelingContext } from 'hooks/useModelingContext'
+import { useKclContext } from 'lang/KclProvider'
+import { codeRefFromRange, getArtifactFromRange } from 'lang/std/artifactGraph'
+import { sourceRangeFromRust } from 'lang/wasm'
+import {
+  filterOperations,
+  getOperationIcon,
+  getOperationLabel,
+} from 'lib/operations'
+import { editorManager, engineCommandManager, kclManager } from 'lib/singletons'
+import { ComponentProps, useEffect, useMemo, useRef, useState } from 'react'
+import { Operation } from 'wasm-lib/kcl/bindings/Operation'
+import { Actor, Prop } from 'xstate'
+import { featureTreeMachine } from 'machines/featureTreeMachine'
+import {
+  editorIsMountedSelector,
+  kclEditorActor,
+  selectionEventSelector,
+} from 'machines/kclEditorMachine'
+
+export const FeatureTreePane = () => {
+  const isEditorMounted = useSelector(kclEditorActor, editorIsMountedSelector)
+  const lastSelectionEventId = useSelector(
+    kclEditorActor,
+    selectionEventSelector
+  )
+  const { send: modelingSend, state: modelingState } = useModelingContext()
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_featureTreeState, featureTreeSend] = useMachine(
+    featureTreeMachine.provide({
+      guards: {
+        codePaneIsOpen: () =>
+          modelingState.context.store.openPanes.includes('code') &&
+          editorManager.editorView !== null,
+      },
+      actions: {
+        openCodePane: () => {
+          modelingSend({
+            type: 'Set context',
+            data: {
+              openPanes: [...modelingState.context.store.openPanes, 'code'],
+            },
+          })
+        },
+        sendEditFlowStart: () => {
+          modelingSend({ type: 'Enter sketch' })
+        },
+        scrollToError: () => {
+          editorManager.scrollToFirstErrorDiagnosticIfExists()
+        },
+        sendSelectionEvent: ({ context }) => {
+          console.log('sendSelectionEvent', context)
```

Comment:
Do we want to leave these console logs in? There's a few in this file.

Reply:
No, no I do not

Reply:
Done